### PR TITLE
 Added  Abstraction in Object Oriented Programming 

### LINKED ISSUE #1195: [Feature]: Add Abstraction in Object Oriented Programming
### Feature Name

Abstraction 

### Feature Description

Abstraction in Java is the process of hiding unnecessary implementation details from the user, focusing on exposing only the essential functionalities. It allows developers to interact with objects at a higher level, without needing to understand the complexities of the underlying code. This ensures that the program is more modular, maintainable, and user-friendly.

### Motivation

Abstraction is one of the core principles of object-oriented programming (OOP). It plays a vital role in simplifying code management, reducing complexity, and enhancing security by hiding implementation details. It is widely used in real-world applications, such as GUI systems, database management, and enterprise-level software, to keep code cleaner and easier to manage.

### Implementation Suggestions (Optional)

_No response_

### Feature Type

New Algorithm

### Does this feature require additional resources?

- [x] Additional Documentation
- [x] Third-Party Libraries
- [x] UI/UX Design
- [ ] Other

### References (Optional)

_No response_   improve it

### PR DESCRIPTION
**Pull request by tanishqkolhatkar93**
### Description
Abstraction in Java is the process of hiding unnecessary implementation details from the user, focusing on exposing only the essential functionalities. It allows developers to interact with objects at a higher level, without needing to understand the complexities of the underlying code. This ensures that the program is more modular, maintainable, and user-friendly.

Fixes #1195

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tanishqkolhatkar93** [comment]
@ajay-dhangar please review and merge this PR

File: docs/Object Oriented Programming/Abstraction.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
+# Abstraction
```
Comment (ajay-dhangar) [inline docs/Object Oriented Programming/Abstraction.md]: please follow our formate and carefully read our contributions guidelines for better understanding our structures

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by tanishqkolhatkar93** [comment]
@ajay-dhangar I have made the requested changes please review it

File: docs/Object Oriented Programming/Abstraction.md [hunk 2]
```diff
@@ -0,0 +1,394 @@
+id: Abstraction
+title: Abstraction in Object Oriented Programming
+sidebar_label: Generate Detail about Abstraction
+description: "Abstraction in Java is the process of hiding unnecessary implementation details from the user, focusing on exposing only the essential functionalities. It allows developers to interact with objects at a higher level, without needing to understand the complexities of the underlying code. This ensures that the program is more modular, maintainable, and user-friendly."
+tags: [Abstraaction, OOPS, Java]
+
```
Comment (ajay-dhangar) [inline docs/Object Oriented Programming/Abstraction.md]: write within `---` and you can checkout format in our Contributions guideline

File: docs/Object Oriented Programming/Abstraction.md [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+class Main {
+  public static void main(String[] args) {
+    Cat myCat = new Cat();  // Create a Cat object
+    myCat.animalSound();
+    myCat.sleep();
+  }
+}
+```
+
+### Multiple Interfaces <a name="multiple-interface"></a>
```
Comment (ajay-dhangar) [inline docs/Object Oriented Programming/Abstraction.md]: why write `<a name="multiple-interface"></a>`

File: docs/Object Oriented Programming/Abstraction.md [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+## Multiple inheritance in Java by interface
+
+If a class implements multiple interfaces, or an interface extends multiple interfaces, it is known as multiple inheritance.
+
+<p align="center">
+<img height="350" src="https://user-images.githubusercontent.com/13514156/120522069-b4c32580-c39a-11eb-9c48-6331fe379440.png">
+</p>
+
+```Multiple inheritance is not supported through class in java, but it is possible by an interface, why?```
+
```
Comment (ajay-dhangar) [inline docs/Object Oriented Programming/Abstraction.md]: 😕 ... ⁉️

File: docs/Object Oriented Programming/Abstraction.md [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+public void print(){System.out.println("Hello");}  
+public static void main(String args[]){  
+TestInterface3 obj = new TestInterface3();  
+obj.print();  
+ }  
+} 
+
+// RESULT
+Hello
+```
```
Comment (ajay-dhangar) [inline docs/Object Oriented Programming/Abstraction.md]: ❌

File: docs/Object Oriented Programming/Abstraction.md [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+}	
+```
+
+Example:
+```java
+public interface Drawable{
+void d
+aw();
+}
+
```
Comment (ajay-dhangar) [inline docs/Object Oriented Programming/Abstraction.md]: ⁉️

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please update your code to follow the rules in our documentation. This includes using good practices, syntax, and formatting.

**Comment by tanishqkolhatkar93** [comment]
@ajay-dhangar I have made the changes please review it

**Review by ajay-dhangar — APPROVED** [review]
Done

**Review by ajay-dhangar — APPROVED** [review]
👍

**Pull request merged**